bit_manipulation/largest_pow_of_two_less_than_or_equal_to_a_number.py

```diff
@@ -0,0 +1,60 @@
+"""
+Author  : Naman Sharma
+Date    : October 2, 2023
+
+Task:
+Find largest power of 2 less than or equal to a given number
+
+Implementation notes: Use bit manipulation.
+We start from 1 & left shift the set bit to check if (res<<1)<=number.
+Each left bit shift represents a pow of 2.
+
+For example:
+number: 15
+res:    1   0b1
+        2   0b10
+        4   0b100
+        8   0b1000
+        16  0b10000 (Exit)
+"""
+
+
```

Comment:
i dont think we need so long one instead we can use 
def find(a):
    if a<=0:
        return 0;
    for i in range(a+1):
        if (2**i <= a):
            c=i;
        else :
            return c;
b=int(input())
print(find(b))

Reply:
Hey @rahbdd thanks for the suggestion but it's actually not about the size of the code but how verbose yet briefly it explains the algo.  The actual code I have written is in fact only 5 lines.

I would suggest you to look at all the other algo's in the repository. The goal of the **TheAlgorithms** as described by their authors is _to document model beautiful, helpful and interesting algorithms using code_

Reply:
Thank you , I will try to modify it